src/py/flwr/client/client_app.py

```diff
@@ -32,19 +32,21 @@
 
 def _inspect_maybe_adapt_client_fn_signature(client_fn: ClientFnExt) -> ClientFnExt:
     client_fn_args = inspect.signature(client_fn).parameters
+    first_arg = list(client_fn_args.keys())[0]
 
-    if not all(key in client_fn_args for key in ["node_id", "partition_id"]):
+    if len(client_fn_args) != 1 or client_fn_args[first_arg].annotation is not Context:
```

Comment:
The first arg might not have a `Context` annotation - we want users to allow `def client_fn(context):`, right?

Maybe we should just check the name: if the name is `cid`, or if it has a type annotation `str`, we use compat mode. Otherwise we assume `Context`.

Reply:
What should in the case where we have a type annotation, but it's neither `str` or `Context`?